hypothesis/lms
# Moodle course copy fixes

## Testing 

- Login in moodle as `moodleteacher`.
- The course copy fix relies on storing data in the original course. Simulate that using the "Select content" button in  [HTML - Group Assignment](https://hypothesisuniversity.moodlecloud.com/course/modedit.php?update=861&return=1)
- Select any type of content and the checkbox for "This is a group assignment". Do not save the assignment. We are only after the side effect.
- Launch [
HTML - Group Assignment](https://hypothesisuniversity.moodlecloud.com/mod/lti/view.php?id=861)

- [Launch the equivalent assignment on the copied course](https://hypothesisuniversity.moodlecloud.com/mod/lti/view.php?id=887)

- Check the DB for the state there:

````
select lms_name, extra from grouping where type = 'course';
-[ RECORD 2 ]-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
lms_name | Developers test course
extra    | {"group_sets": [{"id": "2", "name": "Grouping in course"}, {"id": "5", "name": "Grouping without  groups"}]}
-[ RECORD 3 ]-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
lms_name | (Copied-2024) Developer tests course
extra    | {"group_sets": [{"id": "6", "name": "Grouping in course"}, {"id": "7", "name": "Grouping without  groups"}],  "course_copy_group_set_mappings": {"2": "6"}}
```

Note the course_copy_group_set_mappings in the copied course with points the orignal group set (2) to the equivalent new one (6).
